app/file.go

```diff
@@ -545,28 +545,24 @@ type UploadFileTask struct {
 
 func (t *UploadFileTask) init(a *App) {
 	t.buf = &bytes.Buffer{}
-	t.maxFileSize = *a.Config().FileSettings.MaxFileSize
-	t.limit = *a.Config().FileSettings.MaxFileSize
+	if t.ContentLength > 0 {
+		t.limit = t.ContentLength
+	} else {
+		t.limit = t.maxFileSize
+	}
+
+	if t.ContentLength > 0 && t.ContentLength < maxUploadInitialBufferSize {
+		t.buf.Grow(int(t.ContentLength))
+	} else {
+		t.buf.Grow(maxUploadInitialBufferSize)
```

Comment:
To verify my understanding - most uploads come with mime/multipart, so no content-length; so in real life we are always going to be starting with a 1Mb buffer/file.

Reply:
True. However this might change soon. I am currently working on resumable uploads support which will most probably not use multipart but go straight into this code-path.